### PR TITLE
Add PyJWKSet

### PR DESCRIPTION
**Pull request by jquense**
Adds a new class for managing JWK sets. Let me know if this is moving in a direction you think makes sense. I can add comments and doc bits if this seems more or less like something you'd want to take

### CODE REVIEW

**Comment by chirag04** [comment]
cc @jpadilla

File: setup.cfg [hunk 1]
```diff
@@ -7,8 +7,8 @@ exclude =
 [wheel]
 universal = 1
 
-[tool:pytest]
-addopts = --cov-report term-missing --cov-config=.coveragerc --cov .
+; [tool:pytest]
+; addopts = --cov-report term-missing --cov-config=.coveragerc --cov .
```
Comment (wichert) [inline setup.cfg]: Why are you commenting this out?

**Comment by wichert** [comment]
Looking at the changes I don't see how one would use the set?

**Comment by jquense** [comment]
i'm not sure i understand the question? Do the tests not demostrate usage, or you mean in more of a practical sense "I don't understand why someone would use this"?

**Comment by jgehrcke** [comment]
I think this deserves a proper review. Superficially it looks nice.

**Pull request closed without merging**